File: 백준/Silver/11726. 2×n 타일링/2×n 타일링.cc
#include <iostream>

using namespace std;

int d[1002];

int mod = 10007;
int main()
{
    int n;
    cin >> n;
    d[1] = 1;
    d[2] = 2;
    for(int i = 3; i <= n; i++){
        d[i] = (d[i-2] + d[i-1]) % mod;
    }
    
    cout << d[n];
    return 0;
}

File: 백준/Silver/11726. 2×n 타일링/2×n 타일링.java
import java.io.FileInputStream;

import java.util.Scanner;

public class Main {

	public static void main(String[] args) throws Exception {
		Scanner sc = new Scanner(System.in);

		int N = sc.nextInt();
		long MOD = 10007;
		long dp[] = new long[N + 2];

		dp[1] = 1;
		dp[2] = 2;

		for (int i = 3; i <= N; i++) {
			dp[i] = dp[i - 1] % MOD + dp[i - 2] % MOD;
		}

		System.out.println(dp[N] % MOD);

	}
}